src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -49,82 +50,137 @@ final class AddCartRuleToOrderHandler extends AbstractOrderHandler implements Ad
     public function handle(AddCartRuleToOrderCommand $command): void
     {
         $order = $this->getOrderObject($command->getOrderId());
+        $orderInvoice = $this->createSingleOrderInvoice($order, $command);
 
-        // If the discount is for only one invoice
-        if ($order->hasInvoice() && null !== $command->getOrderInvoiceId()) {
-            $orderInvoice = new OrderInvoice($command->getOrderInvoiceId()->getValue());
-            if (!Validate::isLoadedObject($orderInvoice)) {
-                throw new OrderException('Can\'t load Order Invoice object');
+        $discountValue = (float) str_replace(',', '.', $command->getDiscountValue());
+        $cartRules = $this->buildCartRulesByType($command, $order, $discountValue, $orderInvoice);
+
+        $result = true;
+
+        foreach ($cartRules as &$cartRule) {
+            $cartRuleObj = $this->createCartRule($command, $order, $cartRule, $discountValue);
+
+            if (false === $cartRuleObj->add()) {
+                break;
+            }
+
+            $cartRule['id'] = $cartRuleObj->id;
+        }
+
+        if ($result) {
```

Comment:
$result is always true here :thinking:

Reply:
Fixed above:
```
            if ($result = $cartRuleObj->add()) {
                $cartRule['id'] = $cartRuleObj->id;
            } else {
                break;
            }
```
Original stmt from previous pr.